Maple2.Server.Web/Controllers/BlueprintController.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.IO;
+using Maple2.Tools;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Maple2.Server.Web.Controllers;
+
+[Route("/blueprint/ms2/01/")]
+public class BlueprintController : ControllerBase {
+
+    [HttpGet("{blueprintId}/{ugcUid}.png")]
+    public IResult GetBlueprint(long blueprintId, string ugcUid) {
```

Comment:
**🚨 suggestion (security):** Consider improving file path handling and error responses

1. Sanitize or validate the `blueprintId` and `ugcUid` inputs to prevent potential path traversal attacks.
2. Consider using a configuration value for the base path (`Paths.WEB_DATA_DIR`) instead of hardcoding it.
3. Implement more specific error handling, such as distinguishing between 'not found' and 'unauthorized access' scenarios.
4. Consider using a more specific return type instead of `IResult` for better type safety and API documentation.